tests/unit_tests/oci/test_cloud.py

```diff
@@ -0,0 +1,222 @@
+        oci_cloud.compute_client.list_images.return_value = mock.Mock(data=[])
+        with pytest.raises(PycloudlibException, match="Image with name test-image not found"):
+            oci_cloud.get_image_id_from_name("test-image")
+
+    @mock.patch("pycloudlib.oci.cloud.wait_till_ready")
+    def test_snapshot(self, mock_wait_till_ready, oci_cloud):
+        """Test snapshot method."""
+        instance = mock.Mock()
+        oci_cloud.compute_client.create_image.return_value = mock.Mock(
+            data=mock.Mock(id="image-id")
+        )
+        mock_wait_till_ready.return_value = mock.Mock()
+
+        oci_cloud.snapshot(instance, clean=True, name="snapshot-name")
+        instance.clean.assert_called_once()
+        oci_cloud.compute_client.create_image.assert_called_once()
+
+
+class TestOciInstances:
+    """Tests for OCI Cloud instance-related functions."""
+
+    @mock.patch("oci.config.from_file")
+    def test_get_instance(self, mock_from_file, oci_cloud):
+        """Test get_instance method with valid instance."""
+        valid_config = {
+            "user": "ocid1.user.oc1..example",
+            "fingerprint": "mock-fingerprint",
+            "key_file": "/path/to/key",
+            "tenancy": "ocid1.tenancy.oc1..example",
+            "region": "us-phoenix-1",
+        }
+        # Mock oci.config.from_file to return the valid configuration
+        mock_from_file.return_value = valid_config
+
+        oci_cloud.compute_client.get_instance.return_value = mock.Mock()
+        instance = oci_cloud.get_instance("test-instance-id", username="opc")
+        assert isinstance(instance, OciInstance)
+        oci_cloud.compute_client.get_instance.assert_called_once_with("test-instance-id")
+
+    @mock.patch("oci.config.from_file")
+    def test_get_instance_not_found(self, mock_from_file, oci_cloud):
+        """Test get_instance raises InstanceNotFoundError when instance does not exist."""
+        valid_config = {
+            "user": "ocid1.user.oc1..example",
+            "fingerprint": "mock-fingerprint",
+            "key_file": "/path/to/key",
+            "tenancy": "ocid1.tenancy.oc1..example",
+            "region": "us-phoenix-1",
+        }
+        mock_from_file.return_value = valid_config
+
+        oci_cloud.compute_client.get_instance.side_effect = oci.exceptions.ServiceError(
+            status=404,
+            code="NotFound",
+            message="Instance not found",
+            headers={},
+            target_service=None,
+        )
+        with pytest.raises(InstanceNotFoundError):
+            oci_cloud.get_instance("test-instance-id")
+
+    @mock.patch("pycloudlib.oci.cloud.wait_till_ready")
+    def test_launch_instance(self, mock_wait_till_ready, oci_cloud):
+        """Test launch method with valid inputs."""
+        # mock the key pair
+        oci_cloud.key_pair = mock.Mock()
+        oci_cloud.key_pair.public_key_content = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQC"
```

Comment:
Can squash it to one line w/ our 100 char limit
```suggestion
        oci_cloud.key_pair = mock.Mock(public_key_content="ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQC")
```